firefox-ios/Client/Ecosia/Extensions/BrowserViewController+Ecosia.swift

```diff
@@ -87,8 +93,74 @@ extension BrowserViewController {
     private func showLoadingScreen(for user: User) -> Bool {
         user.referrals.pendingClaim != nil
     }
+}
+
+// MARK: Present insightful sheets
+extension BrowserViewController {
+    private var shouldShowDefaultBrowserPromo: Bool {
+        profile.prefs.intForKey(PrefsKeys.IntroSeen) == nil &&
+        DefaultBrowser.minPromoSearches <= User.shared.searchCount
+    }
+    private var shouldShowWhatsNewPageScreen: Bool { whatsNewDataProvider.shouldShowWhatsNewPage }
+
+    func presentInsightfulSheetsIfNeeded() {
+        guard isHomePage(),
+              !showLoadingScreen(for: .shared) else { return }
+
+        // TODO: To review this logic as part of the upgrade
```

Comment:
I made a [JIRA Ticket](https://ecosia.atlassian.net/browse/MOB-3188) to specifically tackle this TODO

Reply:
Thank you! Did not fully remember what we meant by that, good that it came up 🙂